src/tyr/route_serializer_osrm.cc

```diff
@@ -1062,7 +1062,8 @@ summarize_leg(google::protobuf::RepeatedPtrField<valhalla::DirectionsLeg>::const
 
   // Create a list of named segments (maneuver name, index, distance items)
   std::vector<NamedSegment> named_segments;
-  for (const auto map_item : maneuver_summary_map) {
+  named_segments.reserve(maneuver_summary_map.size());
```

Comment:
It's pretty cool that clang-tidy deduced that it could introduce this `reserve` to optimize memory allocations.